plugins/pagerduty-incidents/src/components/PagerDutyPicker.tsx

```diff
@@ -0,0 +1,169 @@
+import type React from "react";
+import { useCallback, useState, useEffect, useMemo } from "react";
+import "../baseStyles.css";
+
+import { usePluginContext } from "@cortexapps/plugin-core/components";
+
+import { Flex, Button, Select } from "@chakra-ui/react";
+
+import {
+  // Document,
+  parseDocument,
+} from "yaml";
+
+interface PagerDutyPickerProps {
+  entityYaml: Record<string, any>;
+  changed: () => void;
+}
+
+const PagerDutyPicker: React.FC<PagerDutyPickerProps> = ({ changed }) => {
+  const context = usePluginContext();
+  const entityTag: string = useMemo(
+    () => context.entity?.tag ?? "",
+    [context.entity?.tag]
+  );
+  const [services, setServices] = useState([] as any[]);
+  const [serviceSelectOptions, setServiceSelectOptions] = useState([] as any[]);
+  const [selectedService, setSelectedService] = useState("" as any);
+  const [isLoading, setIsLoading] = useState(false);
+  const [fetchedEntityDocument, setFetchedEntityDocument] = useState(
+    null as any
+  );
+
+  const updateEntity = useCallback((): void => {
+    const doUpdate = async (): Promise<void> => {
+      const newDoc = fetchedEntityDocument.clone();
+      const info = newDoc.get("info");
+      info.set("x-cortex-oncall", {
+        pagerduty: {
+          id: selectedService,
+          type: "SERVICE",
+        },
+      });
+
+      const url = `${context.apiBaseUrl}/open-api`;
+      const response = await fetch(url, {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/openapi;charset=utf-8",
+        },
+        body: newDoc.toString(),
+      });
+      if (!response.ok) {
+        console.error(`HTTP Error ${response.status}: ${response.statusText}`);
+        return;
+      }
+      changed();
+    };
+    void doUpdate();
+  }, [context.apiBaseUrl, fetchedEntityDocument, selectedService, changed]);
+
+  useEffect(() => {
+    const fetchServices = async (): Promise<void> => {
```

Comment:
I would make the same comment here as in PagerDutyIncidents, that we can reduce complexity by fetching with custom hooks, this one has less UI components though so I think we are ok

Reply:
i put all the PD fetch effects into hooks but left the cortex ones in as effects, let me know what you think